gpt_engineer/core/ai.py

```diff
@@ -18,7 +18,8 @@
     messages_from_dict,
     messages_to_dict,
 )
-from langchain_community.chat_models import AzureChatOpenAI, ChatOpenAI
+from langchain_community.chat_models import AzureChatOpenAI
```

Comment:
When looking up [docs for langchain_community.chat_models.AzureChatOpenAI](https://api.python.langchain.com/en/latest/chat_models/langchain_community.chat_models.azure_openai.AzureChatOpenAI.html#) they state that it is deprecated.

The [langchain-openai README](https://pypi.org/project/langchain-openai/) state that you should use:

```
from langchain_openai import AzureOpenAI
```

Reply:
So i just looked into making this change. While it seems simple enough I have no way to test it works as i dont have an azure account with an azure open ai set up or the time to set this up. Its possible the API of AzureOpenAI is different to AzureChatOpenAI, and without actually being able to verify the new code works id rather not make this change as part of this PR - but i can create a seperate ticket for it?

Reply:
Oops, I wrote `AzureOpenAI` but meant `AzureChatOpenAI` (edited my comment).

I'm fine with it either way, but I think this is a safe change to make without testing.

I have an Azure account that I could test with, but feel free to wait with that for another PR :)

Reply:
Ah cool I was looking at azure so . I'll do this then

Reply:
Done [3e83dfc](https://github.com/gpt-engineer-org/gpt-engineer/pull/984/commits/3e83dfc2cf963cd9f127276acd4b08193a6c19df)